README.md

```diff
@@ -4,30 +4,28 @@
 
 ### TL;DR
 
-We use this within our components to publish [statsd](http://github.com/etsy/statsd) metrics.
+We use this within our components to publish [statsd](http://github.com/etsy/statsd) metrics from .Net code.
 
-### In detail
-
-There's really not much else to say :-)
-
-We've been using this in most of our things, mostly unchanged, since around 2013.
+We've been using this in most of our things, since around 2013.
 
 ### Features
 
 * statsd metrics formatter
 * UDP client handling
 
-####
+#### Publishing statistics
+
+`IStatsDPublisher` is the interface that you will use in most circumstances. With this you can `Increment` or `Decrement` an event, and send values for a `Gauge` or `Timing`.
 
-`IStatsDPublisher` is the interface that you will use in most circumstances. Use an instance of this in order to send events, timers and gauges.
+The concrete class that implements `IStatsDPublisher` is `StatsDImmediatePublisher`. For the constructor parameters, you will need the statsd server host name. You can change the standard port (8125). You can also append a prefix to all stats. These values often come from configuration as the host name and/or prefix may vary between test and production environments.
```

Comment:
Good spot, I have fixed it.

If you want to add another config option for a stat suffix then you would need another constructor parameter.I'm not a great fan of injecting primitive types such as these `string` and `int` using an IoC container. It's a less-used feature of the various containers, meaning that support and syntax is going to vary between different containers; developer familiarity will be lower etc. 

So when you find yourself doing more of it, there may be a better way. i.e. making a type to hold all these values, so that programmer and IoC container have a unique type to work with. It also allows for sensible defaults.
e.g. 
```csharp
public class StatsDSettings
{
   public int Port { get; set; }
   public string Host { get; set; }
  // etc
}

public class StatsDPublisher
{
   public StatsDPublisher(StatsDSettings settings)
   {
   // etc
  }
}

```